mbf_abstract_nav/src/abstract_planner_execution.cpp

```diff
@@ -261,47 +256,48 @@ void AbstractPlannerExecution::run()
         current_tolerance = tolerance_;
         ROS_INFO_STREAM("A new goal pose is available. Planning with the new goal pose and the tolerance: "
                         << current_tolerance);
-        exceeded = false;
         geometry_msgs::Point g = goal_.pose.position;
         ROS_INFO_STREAM("New goal pose: (" << g.x << ", " << g.y << ", " << g.z << ")");
       }
 
-      make_plan = !(success || exceeded) || has_new_start_ || has_new_goal_;
```

Comment:
make_plan is actually redundant, since we would set planning_ to false in all but the 'retry' case. Also in the old implementation we had an unhandled else case (which could never happen, but still) if which cancel_ and make_plan would be false. In this case we would go for busy spinning at full speed